alchemist/alchemist-incarnation-biochemistry/src/main/java/it/unibo/alchemist/model/implementations/actions/CellTensionPolarization.java

```diff
@@ -0,0 +1,141 @@
+        }
+    }
+
+    @Override
+    public CellTensionPolarization cloneOnNewNode(final Node<Double> n, final Reaction<Double> r) {
+        return new CellTensionPolarization(n, env);
+    }
+
+    @Override
+    public void execute() {
+        // initializing resulting versor
+        double[] resVersor = new double[2];
+        // get node position as array
+        final double[] nodePos = env.getPosition(getNode()).getCartesianCoordinates();
+        // transforming each node around in a vector (Position) 
+        final List<Position> pushForces = env.getNodesWithinRange(
+                getNode(),
+                env.getMaxDiameterAmongDeformableCells()).stream()
+                .parallel()
+                .filter(n -> { // only cells overlapping this cell are selected
+                    if (n instanceof CellWithCircularArea) {
+                        // computing for each cell the max distance among which cant't be overlapping
+                        double maxDist;
+                        if (n instanceof CircularDeformableCell) {
+                            // for deformable cell is maxRad + maxRad
+                             maxDist = (((CircularDeformableCell) getNode()).getMaxRadius() + ((CircularDeformableCell) n).getMaxRadius());
+                        } else {
+                            // for simple cells is maxRad + rad
+                             maxDist = (((CircularDeformableCell) getNode()).getMaxRadius() + ((CellWithCircularArea) n).getRadius());
+                        }
+                        // check
+                        return env.getDistanceBetweenNodes(getNode(), n) < maxDist;
+                    } else {
+                        // only CellWithCircularArea are selected.
+                        return false;
+                    }
+                })
+                .map(n -> {
+                    // position of node n as array
+                    final double[] nPos =  env.getPosition(n).getCartesianCoordinates();
+                    // max radius of n
+                    final double maxRn;
+                    // min radius of n
+                    final double minRn;
+                    // max radius of this node (getNode())
+                    final double maxRN = ((CircularDeformableCell) getNode()).getMaxRadius();
+                    // min radius of this node (getNode())
+                    final double minRN = ((CircularDeformableCell) getNode()).getRadius();
+                    // intensity of tension between n and this node (getNode()), measured as value between 0 and 1
+                    final double intensity;
+                    if (n instanceof CircularDeformableCell) {
+                        maxRn = ((CircularDeformableCell) n).getMaxRadius();
+                        minRn = ((CircularDeformableCell) n).getRadius();
+                    } else {
```

Comment:
Sorry, didn't understand that. Isn't it checked? I verify if n is an instance of CircularDeformableCell and then I cast it.